chia-client/src/peer.rs

```diff
@@ -165,20 +178,20 @@ impl Peer {
         // Remove the one shot channel.
         self.requests.lock().await.remove(&message_id);
 
-        match response {
-            Ok(message) => {
+        // Handle the response, if present.
+        response
+            .map(|message| {
                 let expected_type = R::msg_type();
                 let found_type = message.msg_type;
 
                 if found_type != expected_type {
-                    return Err(Error::InvalidResponse(Some(message)));
+                    return Err(Error::InvalidResponse(message));
                 }
 
                 R::parse(&mut Cursor::new(message.data.as_ref()))
```

Comment:
the `Streamable` trait has a new `from_bytes()` member function as a short-hand for this now

Reply:
Nice